Proper rounding implementation

davidfoerster opened the pull request:
…in accordance with Basisregelwerk page 13 (“Rundungen”).

The previous implementation relies on `string.format("%.0f", x)` whose behaviour is inconsistent across Lua versions.

Other rounding uses idioms like

 * `math.floor(x + 0.5)` which is fine for DSA number ranges but generally inexact,
 * `(n + d2) // d` (where `d` and `d2` are positive integer literals and `d2` is (`d`÷2) rounded down) which is a bit difficult to read and error-prone.

Potential caveat: this implementation rounds halfway cases away from zero while the rules state that (barring special cases) they should be rounded “up”, i. e. towards positive infinity. However, I have yet to see a rule that requires the rounding of *negative* values (to any other value than zero); instances where negative modifiers need rounding (e. g. encumbrance), the rounding is supposed to be applied to their absolute values, i. e. away from zero.

Commit:
Add assert() around loadfile() invocations

Commit:
Implement math.round(x): nearest integer, halfway away from zero

...in accordance with Basisregelwerk page 13 ("Rundungen").

The previous implementation relies on `string.format("%.0f", x)` whose
behaviour is inconsistent accross Lua versions.

Commit:
Use math.round() over common.round() in trivial cases

Commit:
common.render_delta(): use math.round(), fix other issue

- common.render_delta() used to generate the wrong sign for non-zero
  values in [-0.5, 0.5].
- Use local variable for sign character to better describe what's
  happening.

Commit:
kampfbogen.energieleiste(): use math.round(), fix other issue

- Use a *local* variable to store intermediary results.
- Emit "&" and the integral value with a single tex.sprint() since the latter
  can't result in special characters.

Commit:
Remove common.round()

Commit:
Convert other instances to math.round()

that previously involved `math.floor(x + 0.5)` or `string.format("%.0f", x)`.

flyx commented on src/kampfbogen.lua:
```diff
@@ -252,7 +253,7 @@ local waffenlos_render = {
       end
     end
-    tex.sprint(-2, atpa_mod(atb - common.round(ebe/2, true), talent.AT, v["TP/KK Schwelle"], v["TP/KK Schritt"], 0))
+    tex.sprint(-2, atpa_mod(atb - math.round(ebe/2, true), talent.AT, v["TP/KK Schwelle"], v["TP/KK Schritt"], 0))
```
Selbe Geschichte wie oben.

flyx reviewed (changes requested):
Generell okay, da war schon viel Wildwuchs in den Rundungsmechanismen. 

Allerdings zieht ungerade eBE jetzt einen Punkt zu viel von der AT ab, das müsstest du noch fixen.

davidfoerster commented:
Danke für die Kommentare! Ich kümmere mich im Verlauf der Woche um die beiden Probleme.

Commit:
Kampfbogen: Fix "rounding" of effective AT/PA encumberance

Strictly speaking, the rules of distribution of effective encumberance onto
AT/PA values use integer floor/ceil division by two rather than rounding
(Basisregelwerk, p. 137). So that's what we'll use in the implementation.

davidfoerster commented:
@flyx Bitteschön.
